src/CodeGen_LLVM.cpp

```diff
@@ -2939,7 +2939,18 @@ void CodeGen_LLVM::visit(const Call *op) {
         internal_assert(op->args.size() == 2);
         Expr x = op->args[0];
         Expr y = op->args[1];
-        Expr e = Internal::halide_exp(Internal::halide_log(x) * y);
+        Halide::Expr abs_x_pow_y = Internal::halide_exp(Internal::halide_log(abs(x)) * y);
+        Halide::Expr nan_expr = make_const(x.type(), nan(""));
```

Comment:
I'm not sure if we handle nans correctly as float constants. Use Call::make(type, "nan_f32", {}, Call::PureExtern);

Reply:
I had `Call::make(type, "nan_f32", {}, Call::PureExtern);` originally but changed it to constant float to make windows/cuda happy. let see if this will work with CSE applied.